explorations/s3StructureSketch.R

```diff
@@ -0,0 +1,152 @@
+# Note: This is very incomplete and somewhere between R and pseudocode
+
+develop <- function(x, ... ) UseMethod("develop", x)
+
+develop.formula <- function(formula, data, ...) {
+  
+  if (!inherits(formula, "formula")) 
+    stop("develop.formula needs a y ~ x formula as its first argument.  
+             You passed it a ", class(formula))
+  
+  # Parse the formula to get the data and then use standard generic:
+  develop(data, predictColumn, ... )
+}
+
+# develop is already way too long and therefore non-modular.
+# Needs to be chunked out into smaller functions.
+develop <- function(data
+                    , predictColumn = NULL
+                    , type = NULL
+                    # Default to using all algorithms defined for the type:
+                    , algorithms = NULL
+                    # It's a little weird to put this much info the default, but
+                    # it would make it easy for users to copy this list and modify 
+                    # it; otherwise nested lists can be tricky.
+                    , tuningParams = list(randomForest = list(mtry = c(10, 100, 1000),
```

Comment:
I agree with your first statement - this could easily become a heinous interface.